### PR TITLE
Bug 1169019 - Only speak when screenreader enabled or forced

### PR DESCRIPTION
**Pull request by jobara**
https://bugzilla.mozilla.org/show_bug.cgi?id=1169019

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=645ddb813b8563a678c274c7206dd83b114e5f3c

File: apps/system/test/unit/accessibility_test.js [hunk 1]
```diff
@@ -326,10 +326,16 @@ suite('system/Accessibility', function() {
     test('handle vc change event with hints enabled', function() {
       var stubSetHintsTimeout = this.sinon.stub(accessibility,
         'setHintsTimeout');
+      // Enable screenreader
```
Comment (yzen) [inline apps/system/test/unit/accessibility_test.js]: This better fits in setup and teardown functions?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bc9debb7c570c0fb5fd17c92c52042a702dcf977

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ab4685731ca1e95ccda5f14ac084fd5bb27908f1

File: apps/system/js/accessibility.js [hunk 2]
```diff
@@ -478,8 +471,10 @@
      * @memberof Accessibility.prototype
      */
     speak: function ar_speak(aData, aCallback, aOptions = {}) {
-      this.speechSynthesizer.speak(aData, aOptions, this.rate, this.volume,
-        aCallback);
+      if (this.settings['accessibility.screenreader'] || aOptions.force) {
```
Comment (yzen) [inline apps/system/js/accessibility.js]: Comment would be nice.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7edf22d3c8b0df3db33d7175df109280aec533f4

**Comment by jobara** [comment]
@yzen I've updated the pull request with master. One side effect I'm noticing is when the device is turned on, it will announce that the screen reader is off. Any thoughts on how to avoid this?

File: apps/system/js/accessibility.js [hunk 3]
```diff
@@ -169,6 +169,9 @@
                   this.toggleShade(aValue, !aValue);
                 }
                 this.screen.classList.toggle('screenreader', aValue);
+                this.announceScreenReaderState()
```
Comment (yzen) [inline apps/system/js/accessibility.js]: So lets check aValue against oldValue here and only announce if they are different?

File: apps/system/js/accessibility.js [hunk 4]
```diff
@@ -400,7 +403,6 @@
      * @memberof Accessibility.prototype
      */
     setHintsTimeout: function ar_setHintsTimeout(aHints) {
-      clearTimeout(this.hintsTimer);
       this.hintsTimer = setTimeout(function onHintsTimeout() {
         this.isSpeakingHints = true;
```
Comment (yzen) [inline apps/system/js/accessibility.js]: I think we are getting rid of isSpeakingHints right ?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=09e94ec90be1b957c5b75de06d7c587ab30c768f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=750ef154aa2a6a7b6fdaf9c9506200a4469137f0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=45a88a119723072697c4c7ecc58419ad920959cf

**Comment by jobara** [comment]
@yzen ready for more review

File: apps/system/js/accessibility.js [hunk 5]
```diff
@@ -576,35 +578,41 @@
...
 
     /**
-     * Based on whether the screen reader is currently enabled, announce the
-     * instructions of how to enable/disable it.
+     * Based on whether the screen reader is currently enabled, announce a
+     * message such as instructions of how to enable/disable it.
      * @memberof Accessibility.prototype
      */
-    announceScreenReader: function ar_announceScreenReader() {
+    announceScreenReader: function ar_announceScreenReader(
```
Comment (yzen) [inline apps/system/js/accessibility.js]: TBH: I would rather have the logic outside of this function. So when announceScreenReader is called, check enabled and pass an appropriate key.

Comment (jobara) [inline apps/system/js/accessibility.js]: @yzen "disableScreenReaderSteps" and "enableScreenReaderSteps" are the defaults, but can be overridden by passing in different arguments. Is that what you wanted, or you think there shouldn't be a default value?

**Pull request closed without merging**